models/requests.ts

```diff
@@ -0,0 +1,117 @@
+import { RequestQuery } from "../types/oooRequest";
+import firestore from "../utils/firestore";
+const requestModel = firestore.collection("requests");
+import { REQUEST_ALREADY_APPROVED, REQUEST_ALREADY_REJECTED, REQUEST_STATE } from "../constants/requests";
+import { 
+  ERROR_WHILE_FETCHING_REQUEST,
+  ERROR_WHILE_CREATING_REQUEST,
+  ERROR_WHILE_UPDATING_REQUEST,
+  REQUEST_DOES_NOT_EXIST,
+  REQUEST_ALREADY_PENDING
+} from "../constants/requests";
+
+export const createRequest = async (body: any) => {
+  try {
+    const existingRequest = await requestModel
+      .where("requestedBy", "==", body.requestedBy)
+      .where("state", "==", REQUEST_STATE.PENDING)
+      .where("type", "==", body.type)
+      .get();
+
+    if (!existingRequest.empty) {
+      return {
+        error: REQUEST_ALREADY_PENDING,
+      };
+    }
+    const requestBody: any = {
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+      ...body,
+    };
+    const result = await requestModel.add(requestBody);
+
+    return {
+      id: result.id,
```

Comment:
result.id will return undefined. 
you can pass id directly.

Reply:
I don't think it will return undefined. requestModel.add(requestBody) return object with new id when its created

Reply:
It did return undefined, when writing the test cases. Can you please check once? Maybe I would have mistaken.

Reply:
When the document creation fails, the controller redirects to the catch block and returns an error.